rover-apps/science/src/AdafruitSTEMMAModule.cpp

```diff
@@ -0,0 +1,39 @@
+#include "AdafruitSTEMMAModule.h"
+
+#include <ratio>
+
+#include "Module.h"
+#include "mbed.h"
+
+AdafruitSTEMMAModule::AdafruitSTEMMAModule(PinName sda, PinName scl) : moistureSensor(sda, scl) {
+  int numFails = 0;
+  bool resetStatus    = false;
+
+  while (!resetStatus && numFails < 10) {
+    resetStatus = moistureSensor.reset();
+    numFails++;
+  }
+  // if still error then log through printf (currently), plan to later log through LOGGER (TODO)
+  
+  if (numFails == 10) {
```

Comment:
What if `moistureSensor.reset()` succeeds on the last try but `numFails` then gets incremented to 10?

Reply:
Just an idea but you can add a private bool to indicate whether the sensor was successfully initialized. Then only perform the polling if it that succeeded